modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java

```diff
@@ -150,15 +149,30 @@ private MuleArtifactContext doCreateApplicationContext(MuleContext muleContext,
     if (enableLazyInit) {
       return new LazyMuleArtifactContext(muleContext, resolveArtifactConfigResources(), artifactDeclaration,
                                          optionalObjectsController,
-                                         getArtifactProperties(), artifactType, resolveContextArtifactPluginClassLoaders(),
+                                         getArtifactProperties(), artifactType, resolveAppAndDomainPluginClassLoaders(),
                                          resolveParentConfigurationProperties(), disableXmlValidations);
     }
 
     return new MuleArtifactContext(muleContext, resolveArtifactConfigResources(), artifactDeclaration, optionalObjectsController,
-                                   getArtifactProperties(), artifactType, resolveContextArtifactPluginClassLoaders(),
+                                   getArtifactProperties(), artifactType, resolveAppAndDomainPluginClassLoaders(),
                                    resolveParentConfigurationProperties(), disableXmlValidations);
   }
 
+  private List<ClassLoader> resolveAppAndDomainPluginClassLoaders() {
+    List<ClassLoader> artifactContextPluginClassLoaders = new ArrayList<>(resolveContextArtifactPluginClassLoaders());
+    if (artifactType.equals(ArtifactType.APP) && parentContext != null) {
```

Comment:
If I remember correctly, this check is related to a previous bug you found. Add the jira here

Reply:
Actually, the logic is wrong. We should do this when the parentContext is not null.

Reply:
Actually, the logic is wrong. I should do this for when parentContext is not null